pkg/installer/Dockerfile

```diff
@@ -1,6 +1,33 @@
+# syntax=docker/dockerfile:1.10.0
 # Copyright (c) 2024 Zededa, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
+# Interactive mode for the installer
+FROM --platform=$BUILDPLATFORM lfedge/eve-rust:1.80.1 AS rust-host
+ARG TARGETARCH
+
+# map Docker's $TARGETARCH to Rust target
+FROM rust-host AS target-amd64
+ENV CARGO_BUILD_TARGET="x86_64-unknown-linux-musl"
+
+FROM rust-host AS target-arm64
+ENV CARGO_BUILD_TARGET="aarch64-unknown-linux-musl"
+
+FROM rust-host AS target-riscv64
+ENV CARGO_BUILD_TARGET="riscv64gc-unknown-linux-gnu"
+
+# hadolint ignore=DL3006
+FROM target-$TARGETARCH AS rust
+
+WORKDIR /usr/local/my-installer
+COPY . .
```

Comment:
I would prefer to have an explicit list of what to copy instead of maintaining a blacklist in `.dockerignore`.

Reply:
What we have not it is cleaner and simpler.

Reply:
Fine for me, feel free to resolve, but even in this PR you already forgot to add `.gitignore` (https://github.com/lf-edge/eve/pull/4055/files#r1778545428)

Reply:
I am updating the PR, give me a sec.